python/packages/ai/tests/test_ai.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from unittest import IsolatedAsyncioTestCase
+from unittest.mock import AsyncMock, MagicMock
+from botbuilder.core import TurnContext
+from botbuilder.schema import Activity, ChannelAccount, ConversationAccount
+from teams.state import TurnState
+from teams.ai.ai import AI, AIOptions, ApplicationError
+from teams.ai.actions import ActionEntry, ActionTypes, ActionTurnContext
+from teams.ai.planners.planner import Planner
+from tests.utils import SimpleAdapter
+
+
+class TestAI(IsolatedAsyncioTestCase):
+    def setUp(self):
+        mock_planner = AsyncMock(spec=Planner)
+        options = AIOptions(
+            planner=mock_planner,
+        )
+        self.ai = AI(options)
+
+    def create_mock_context(self):
+        return TurnContext(
+            SimpleAdapter(),
+            Activity(
+                id="1234",
+                type="event",
+                text="test",
+                from_property=ChannelAccount(id="user", name="User Name"),
+                recipient=ChannelAccount(id="bot", name="Bot Name"),
+                conversation=ConversationAccount(id="convo", name="Convo Name"),
+                channel_id="UnitTest",
+                locale="en-uS",
+                service_url="https://example.org",
+            )
+        )
+
+    def create_mock_state(self):
+        state = MagicMock(spec=TurnState)
+        return state
+
+    async def test_do_action_existing_action(self):
+        context = self.create_mock_context()
+        state = self.create_mock_state()
+
+        mock_handler = AsyncMock(return_value="action result")
+
+        self.ai._actions["test_action"] = ActionEntry(
```

Comment:
_actions is technically a private class param, could you try registering via the `action` method?